### PR TITLE
[10.0] [MIG] Purchase Request Procurement (purchase_request_procurement)

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Migration of module purchase_request_procurement to 10.0
Depends on #424

### CODE REVIEW

File: purchase_request_procurement/models/procurement_order.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    request_id = fields.Many2one(
+        comodel_name='purchase.request',
+        ondelete='restrict',
+        string='Latest Purchase Request',
+        copy=False,
+    )
+
+    @api.model
+    def _prepare_purchase_request_line(self, purchase_request, procurement):
```
Comment (LoisRForgeFlow) [inline purchase_request_procurement/models/procurement_order.py]: why not to remove procurement from all the inputs and use self instead?

File: purchase_request_procurement/models/procurement_order.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            'name': procurement.product_id.name,
+            'date_required': procurement.date_planned,
+            'product_uom_id': procurement.product_uom.id,
+            'product_qty': procurement.product_qty,
+            'request_id': purchase_request.id,
+            'procurement_id': procurement.id
+        }
+
+    @api.model
+    def _prepare_purchase_request(self, procurement):
```
Comment (LoisRForgeFlow) [inline purchase_request_procurement/models/procurement_order.py]: comment above

File: purchase_request_procurement/models/procurement_order.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return {
+            'origin': procurement.origin,
+            'company_id': procurement.company_id.id,
+            'picking_type_id': procurement.rule_id.picking_type_id.id,
+        }
+
+    @api.model
+    def _search_existing_purchase_request(self, procurement):
```
Comment (LoisRForgeFlow) [inline purchase_request_procurement/models/procurement_order.py]: same

File: purchase_request_procurement/test/purchase_request_from_procurement.yml [hunk 4]
```diff
@@ -21,11 +21,11 @@
...
-    for procurement in self.browse(cr, uid, procurement_ids, context=context):
+    procurement_ids = self.search([('request_id','!=', False)])
+    for procurement in self.browse(procurement_ids):
         request = procurement.request_id
         assert len(request.line_ids) == 1, "Request Lines should be one."
         line = request.line_ids[0]
-        assert line.product_id.id == procurement.product_id.id, "Product is not correspond."
-        assert line.product_uom_id.id == procurement.product_uom.id, "UOM is not correspond."
-        assert line.product_qty == procurement.product_qty, "Quantity is not correspond."
+        assert line.product_id.id == procurement.product_id.id, "Product does not correspond."
```
Comment (LoisRForgeFlow) [inline purchase_request_procurement/test/purchase_request_from_procurement.yml]: I think there is not goal in mantaining the yml tests in v10. This is a good moment to get rid of this IMO.

**Review by LoisRForgeFlow — COMMENTED** [review]
Tested on runbot and it is fine, just some comments in lines.

**Comment by ThomasBinsfeld** [comment]
@lreficent Thanks for the review. Done !

**Review by LoisRForgeFlow — APPROVED** [review]
Code review of new changes. LGTM now :+1:

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by AaronHForgeFlow** [comment]
Code LGTM. Pending functional tests.

**Comment by LoisRForgeFlow** [comment]
@ThomasBinsfeld Could you check the changes proposed to v9? (see: https://github.com/OCA/purchase-workflow/pull/398 and https://github.com/OCA/purchase-workflow/pull/412)

**Comment by benwillig** [comment]
This PR now depends on #424.

**Comment by ThomasBinsfeld** [comment]
Ping @lreficent

**Comment by LoisRForgeFlow** [comment]
LGTM :+1: 

Just a minor comment, the commit is kind of a cherry-pick of the changes I proposed in v9, is it possible to keep the authority of the commit?

Thanks for the work!

**Comment by benwillig** [comment]
Ups, I forgot to continue my cherry pick :laughing:. It's fixed now.

**Comment by LoisRForgeFlow** [comment]
Thanks! :smile: 

Once the issues with travis are fixed I think this will be ready.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
It fails when a procurement is cancelled :
image

**Comment by benwillig** [comment]
@aheficent At the moment, PR #424 has not been merged yet. That's why it doesn't work.

**Comment by AaronHForgeFlow** [comment]
@benwillig I saw. Let's get that one merged first then. Thank you for telling me.

**Comment by LoisRForgeFlow** [comment]
@ThomasBinsfeld Would you fix conflicts and add this v9 fix made here: https://github.com/OCA/purchase-workflow/commit/08c510912c89489be2c2235e1e96cc6d5826b17e?

edit: the fix was done here: https://github.com/OCA/purchase-workflow/pull/441

**Comment by ThomasBinsfeld** [comment]
Rebased on 10.0

**Comment by LoisRForgeFlow** [comment]
Great, thanks! Just the same that in previous cherry-pick: try to keep the authorship.

**Comment by ThomasBinsfeld** [comment]
@lreficent Authorship restored ! :white_check_mark:

**Comment by D4N** [comment]
#424 has been merged, so could this PR be merged, too? @aheficent Had some comments, but I guess they have been fixed, as #424 is merged.
If there are still issues, I offer my help, as I would really like this module in v10.

**Review by AaronHForgeFlow — APPROVED** [review]
Tested on runbot :+1:

**Comment by JordiBForgeFlow** [comment]
Thanks to all for your efforts!!

**Pull request merged**